sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs

```diff
@@ -295,6 +470,22 @@ private static Response GetResponseFromState(OperationState<T> state)
             throw state.OperationFailedException!;
         }
 
+        private DiagnosticScope CreateScope()
+        {
+            DiagnosticScope scope = _diagnostics.CreateScope(_updateStatusScopeName);
+
+            if (_scopeAttributes != null)
+            {
+                foreach (KeyValuePair<string, string> attribute in _scopeAttributes)
+                {
+                    scope.AddAttribute(attribute.Key, attribute.Value);
+                }
+            }
+
+            scope.Start();
```

Comment:
nit: I'm not wild about this factoring, since it separates the start method from the close method, which makes the code that calls this harder to read and verify at a glance.